src/java.base/share/classes/java/util/SortedSet.java

```diff
@@ -261,4 +261,72 @@ public Comparator<? super E> getComparator() {
             }
         };
     }
+
+    // ========== SequencedCollection ==========
+
+    /**
+     * Not supported. The position of an element is determined by the order induced by
+     * the comparison method, so explicit positioning is not supported.
+     * @implSpec
+     * The implementation in this class always throws {@code UnsupportedOperationException}.
+     * @throws UnsupportedOperationException always
+     * @since 20
+     */
+    default void addFirst(E e) { throw new UnsupportedOperationException(); }
+
+    /**
+     * Not supported. The position of an element is determined by the order induced by
+     * the comparison method, so explicit positioning is not supported.
+     * @implSpec
+     * The implementation in this class always throws {@code UnsupportedOperationException}.
+     * @throws UnsupportedOperationException always
+     * @since 20
+     */
+    default void addLast(E e) { throw new UnsupportedOperationException(); }
+
+    /**
+     * {@inheritDoc}
+     * @since 20
+     */
+    default E getFirst() { return this.first(); }
+
+    /**
+     * {@inheritDoc}
+     * @since 20
+     */
+    default E getLast() { return this.last(); }
+
+    /**
+     * {@inheritDoc}
+     * @since 20
+     */
+    default E removeFirst() {
```

Comment:
Should removeFirst/removeLast be overridden in NavigableSet with a new default implementation like:
```java
    if (this.isEmpty()) {
        throw new NoSuchElementException();
    }
    return this.pollFirst();
```
which is probably slightly more efficient?

Reply:
Good idea, I'll do this now that I'm revisiting the specs of the default methods.